From e420694e99a3f6ecd2342510dfb3f83aa2eb42a6 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 10 Oct 2017 02:07:06 +0100
Subject: [PATCH 1/2] Ensure all libexecdir executables use RPATH

This resolves a linking issue when the libexecdir for tracker-miners differs
from the libexecdir from tracker, i.e. in distributions where each package
has a namespaced libexecdir and not the old-style "/usr/libexec".

Bug-GNOME: https://bugzilla.gnome.org/show_bug.cgi?id=788750

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/tracker-extract/meson.build   | 4 +++-
 src/tracker-writeback/meson.build | 4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/tracker-extract/meson.build b/src/tracker-extract/meson.build
index 2f429e2..7defffb 100644
--- a/src/tracker-extract/meson.build
+++ b/src/tracker-extract/meson.build
@@ -158,7 +158,9 @@ executable('tracker-extract',
   c_args: tracker_c_args + ['-I' + meson.build_root()],
   dependencies: tracker_extract_dependencies,
   install: true,
-  install_dir: join_paths(get_option('prefix'), get_option('libexecdir')))
+  install_dir: join_paths(get_option('prefix'), get_option('libexecdir')),
+  install_rpath: tracker_internal_libs_dir,
+)
 
 dbus_service_file_untranslated = configure_file(
   input: 'org.freedesktop.Tracker1.Miner.Extract.service.in.in',
diff --git a/src/tracker-writeback/meson.build b/src/tracker-writeback/meson.build
index 81aab03..f9f594a 100644
--- a/src/tracker-writeback/meson.build
+++ b/src/tracker-writeback/meson.build
@@ -46,7 +46,9 @@ executable('tracker-writeback',
   ],
   dependencies: [gmodule, tracker_miner, tracker_sparql, tracker_common_dep],
   install: true,
-  install_dir: join_paths(get_option('prefix'), get_option('libexecdir')))
+  install_dir: join_paths(get_option('prefix'), get_option('libexecdir')),
+  install_rpath: tracker_internal_libs_dir,
+)
 
 dbus_service_file = configure_file(
   input: 'org.freedesktop.Tracker1.Writeback.service.in',
-- 
2.14.2

